### PR TITLE
PR for socket blocking in main thread

### PR DESCRIPTION
**Pull request by tvlinh0com**

### CODE REVIEW

File: ex-2/client/client.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+#define BSIZE 1024
+#define PORT 21996
+
+void CleanStdInput(){
+    while(fgetc(stdin) != 10);
+}
+
+void GetStringFromStdInput(char* s, unsigned int length){
+	fgets(s, length, stdin);
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: Wrong alignment.

File: ex-2/client/client.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    int number;
+
+    if (cskid == -1){
+        perror("Create client socket failed");
+        return 0x0;
+    }
+
+    struct sockaddr_in serveraddr = {
+            AF_INET,
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: wrong alignment.

File: ex-2/client/client.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            htons(PORT),
+            htonl(0x7f000001)
+    };
+
+    if (connect(cskid, (struct sockaddr*)&serveraddr, sizeof serveraddr) == -1){
+        perror("Connect to server failed");
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c:56]: Why don't we free the socket and exit here, when the connect failed?

File: ex-2/client/client.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    }else{
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: Put else statement in a separated line.

File: ex-2/client/client.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        printf("Connect to server success!\n");
+        while (!stop){
+            printf("Type to send (%d. Integer Number, %d. String, %d. Student Info): ", NUMBER, STRING, STUDENT_INFO);
+
+            scanf("%d", &type);
+
+            switch (type){
+				case NUMBER:
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: wrong alignment for case.

File: ex-2/client/client.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+					return 0x0;
+					break;
+            }
+
+            printf("Do you want to stop (0: Continue, 1: Stop): ");
+            scanf("%d", &stop);
+        }
+    }
+
+    printf("Closing socket...\n");
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: Let's make a label for EXIT here so that we can goto that label whenever we face error.

File: ex-2/client/client.c [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    close(cskid);
+
+    return 0x0;
```
Comment (nguyenvangiang1988) [inline ex-2/client/client.c]: No need to use hexa here. Just return 0.

File: ex-2/server/server.c [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    if (bind(skid, (struct sockaddr*)&addrport, sizeof addrport) == -1){
+        perror("Bind address to socket failed");
+        return 0x0;
+    }else{
+        printf("Bind port %d success\n", PORT);
+    }
+
+    if(listen(skid, MAXPENDING) == -1){
+        perror("Listening failed");
+        close(skid);
```
Comment (nguyenvangiang1988) [inline ex-2/server/server.c:60]: Let's use "label" and "goto"

File: ex-2/server/server.c [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+                        return 0x0;
+                        break;
+                }
+            }else{
+                printf("Size of type byte incorrect!\n");
+                break;
+            }
+        }
+    }else{
+        perror("accept failed");
```
Comment (nguyenvangiang1988) [inline ex-2/server/server.c:125]: Let's goto the EXIT label when error happens.

File: ex-2/server/server.c [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+
+    printf("Closing socket...\n");
+
+    close(skid);
```
Comment (nguyenvangiang1988) [inline ex-2/server/server.c:131]: Make a EXIT label here

File: ex-2/server/server.c [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+    return 0x0;
```
Comment (nguyenvangiang1988) [inline ex-2/server/server.c]: No need to use hexa number 0x0

File: ex-2/client/client.c (hunk 3 above)
Comment (tvlinh0com) [reply ex-2/client/client.c:56]: Line 120, I free it.

Comment (tvlinh0com) [reply ex-2/client/client.c:56]: Sorry RET main before free it.

**Pull request closed without merging**